src/commands/ipmi.rb

```diff
@@ -36,26 +36,23 @@ class Ipmi < CommandHelpers::BaseCommand
       prepend CommandHelpers::NodeIdentifier
 
       def run
-        if options.group
-          node_names.each do
-            ipmi
-          end
-        else
-          ipmi
+        nodes.each do |node|
+          ipmi(node)
         end
       end
 
-      def ipmi
-        if vm?
-          libvirt_run(libvirt_info[:host], node)
-        else
-          puts "#{node}: #{SystemCommand.run(command(node))}"
-        end
+      def ipmi(node)
+        run_vm(node) if vm?(node)
+        run_baremetal(node) unless vm?(node)
```

Comment:
Splitting out methods for the 2 different `run_*` methods looks good, checking the same condition twice looks a bit odd though - if you want terseness then something like this would be fine:

```ruby
vm?(node) ? run_vm(node) : run_baremetal(node)
```

Reply:
TIL this is a thing.. thanks again